build/xtask/src/dist.rs

```diff
@@ -2524,28 +2519,74 @@ fn write_elf(
     // Page align the start of sections data.
     let sections_data_offset = (program_headers_offset + 0xfff) & !0xfff;
 
+    // Page align the Section Header String Table.
+    let shstrtab_offset =
+        (sections_data_offset + sections_data.len() + 0xfff) & !0xfff;
+
+    // Create a Section Header String Table, to hold the actual section
+    // names.
+    let mut shstrtab = Vec::new();
+    shstrtab.push(0x00 as u8);
+    shstrtab.extend_from_slice(".text".as_bytes());
+    shstrtab.push(0x00 as u8);
+    shstrtab.extend_from_slice(".shstrtab".as_bytes());
+    shstrtab.push(0x00 as u8);
+    shstrtab.shrink_to_fit();
+
+    // Page align the Section Headers.
+    let sh_data_offset = (shstrtab_offset + shstrtab.len() + 0xfff) & !0xfff;
+
+    // We need to create a null section, as without this, objcopy fails to locate
+    // any section headers, and aborts.
+    let mut null_section_header = SectionHeader::new();
+    null_section_header.sh_type = SHT_NULL;
+
+    // We create a fake '.text' section to guide objcopy toward the instruction
+    // bitstream.
```

Comment:
Why not create a separate section for each task and kernel? That would be similar to how the srec->elf translation worked before. Example:

```
Section Headers:
  [Nr] Name              Type             Address           Offset
       Size              EntSize          Flags  Link  Info  Align
  [ 0]                   NULL             0000000000000000  00000000
       0000000000000000  0000000000000000           0     0     0
  [ 1] .sec1             PROGBITS         0000000090000400  00000040
       0000000000010400  0000000000000000  WA       0     0     1
  [ 2] .sec2             PROGBITS         0000000090010800  00010440
       000000000000c900  0000000000000000  WA       0     0     1
  [ 3] .sec3             PROGBITS         000000009001d100  0001cd40
       0000000000001700  0000000000000000  WA       0     0     1
  [ 4] .sec4             PROGBITS         000000009001e800  0001e440
       0000000000002000  0000000000000000  WA       0     0     1
  [ 5] .sec5             PROGBITS         0000000090020800  00020440
       0000000000002e40  0000000000000000  WA       0     0     1
  [ 6] .sec6             PROGBITS         0000000090023640  00023280
       00000000000021c0  0000000000000000  WA       0     0     1
  [ 7] .sec7             PROGBITS         0000000090025800  00025440
       00000000000023c0  0000000000000000  WA       0     0     1
  [ 8] .sec8             PROGBITS         0000000090027bc0  00027800
       0000000000000a28  0000000000000000  WA       0     0     1
  [ 9] .shstrtab         STRTAB           0000000000000000  00028228
       000000000000003b  0000000000000000           0     0     1
```

Reply:
It's an option - happy to go down that path if we think it's preferable. Will want to test the output of objcopy after it's done.

Reply:
I think that could make things slightly simpler to debug when inspecting an ELF image.

Reply:
Added in lasted change.